jaeger_client/metrics.py

```diff
@@ -21,69 +21,155 @@
 from __future__ import absolute_import
 
 
-class Metrics(object):
-    """Provides an abstraction of metrics reporting framework."""
+class MetricsFactory(object):
+    """Generates new metrics."""
+
+    def __init__(self, count=None, gauge=None, timing=None, tags=None):
+        """
+        :param count: function (key, value) to emit counters
+        :param gauge: function (key, value) to emit gauges
+        :param timing: function (key, value) to emit timings
+        :param tags: {k:v} dictionary
+        """
+        self._count = count
+        self._gauge = gauge
+        self._timing = timing
+        self._tags = tags
+        if not callable(self._count):
+            self._count = None
+        if not callable(self._gauge):
+            self._gauge = None
+        if not callable(self._timing):
+            self._timing = None
```

Comment:
what does this achieve? It seems that instead you can default to no-op functions and change the public methods to simply return the `_` values, it would simplify all the derived classes. Or leave the defaulting to the Noop class